### ISSUE TITLE
Suggestion: Node category CSS

### ISSUE THREAD

**Issue by Stumblinbear** [newcat/baklavajs#27]
It may be useful to include a CSS class named `node-<category >` in the parent node div, so that similar nodes can be styled differently. For example: render math nodes with a blue title, smaller in width, or no border radius at all.

**Comment by newcat** [comment]
I'd suggest having a look here and doing it similar to this: https://github.com/newcat/baklavajs/blob/a2c1b0b7ce6859e1563fb2fbc06d23771d9ab8d3/packages/baklavajs-playground/src/App.vue#L57-L62 until this feature is implemented.

**Comment by Stumblinbear** [comment]
Hey that could work, in the mean time. Thanks for the info! I'm still trying to figure out some of the API, as it's not amazingly well documented. ^^

**Comment by newcat** [comment]
Implemented in the `dev` branch. I decided against against using category names but instead, every node div gets a `--type-<NodeType>` modifier. By using CSS selectors you can style all nodes of a certain category at the same type, like this:

```css
.node.--type-NodeA,
.node.--type-NodeB {
    background-color: red;
}
```

**Comment by Stumblinbear** [comment]
Hmm. Question. Can two nodes have the same type? Do they need to be unique? What does type refer to? The documentation is a little... Sparse.. On that subject.

Personally, most of my nodes fit into the category of "logic" "event" or "action" so being able to style them all accordingly without adding each individual one would be a huge boon.

**Comment by newcat** [comment]
Admittedly, I haven't put enough effort into keeping the documentation up to date with the development in the past few months. I will try to improve that. Also just noticed that the API reference is completely broken 🤦‍♂ 

So yeah, a node type is a single entry in the "Add Node" context menu. Basically one kind of node has one type but there can be multiple nodes of the same type. I can see why you would like to have the category in the css class as well. But I have decided against that for the following reason:

The node list is in the [Editor](https://github.com/newcat/baklavajs/blob/master/packages/baklavajs-core/src/editor.ts), which comes from the `core` package. For displaying nodes, the [Editor.vue](https://github.com/newcat/baklavajs/blob/master/packages/baklavajs-plugin-renderer-vue/src/components/Editor.vue), which comes from the `renderer-vue` package, binds to the node list in the editor. The `Editor` (like the rest of the core package) does not "know" the concept of categories. This whole idea is only implemented in the `renderer-vue` package for having a nice and clean context menu. So when rendering the list of nodes, I do not directly have access to the category of a node, as the data comes from the `core` editor. I would have to add this data for each node in the `Editor.vue`, which in turn means that I can't do a simple databinding where all the logic related to a node is separated into a `Node.vue` component.

So TLDR: Although it would be possible to implement, it would blur the line between data sources, which I don't quite like, as it increases complexity and reduces maintainability.

**Comment by Stumblinbear** [comment]
Perhaps then a way to resolve the issue would be to add a hook for context menu creation so that users can alter the entries themselves as necessary. Then, I can use my own internal grouping system and add CSS classes as I see fit. I already do this for nodes themselves, and now connections. Doing it for the context menu doesn't seem like too much of a stretch, at this point.

**Comment by newcat** [comment]
> Edit: It occurs to me I may have posted this in the wrong issue

Ooops and I reopened the wrong issue :)
